### PR TITLE
Unescape query from EMR spark submit parameter

### PR DESCRIPTION
**Pull request by seankao-az**
### Description
In [a previous PR in sql](https://github.com/opensearch-project/sql/pull/2587), query is escaped in spark submit parameter. This PR unescapes the query on the receiving end, retrieving the original query. This affects only the query in SparkConf.

End to end test is performed as follows:
Query:
```
CREATE INDEX cv
ON mys3.default.http_logs (`status`, `day`, `size`, `request`)
WITH (
  auto_refresh = true,
  refresh_interval="5 Minutes",
  checkpoint_location = 's3://test',
  index_settings = "{\"number_of_shards\":1,\"number_of_replicas\":1}"
)
```
Spark submit parameter for the EMR job (query escaped and wrapped with quotes):
```
--conf spark.flint.job.query="CREATE INDEX cv ON mys3.default.http_logs (`status`, `day`, `size`, `request`) WITH (auto_refresh = true, refresh_interval=\"5 Minutes\", checkpoint_location = 's3://test',index_settings = \"{\\\"number_of_shards\\\":1,\\\"number_of_replicas\\\":1}\")"
```
Query received on spark *before* this PR fix (query still escaped and result in syntax error):
```
CREATE INDEX cv ON mys3.default.http_logs (`status`, `day`, `size`, `request`) WITH (auto_refresh = true, refresh_interval=\"5 Minutes\", checkpoint_location = 's3://test',index_settings = \"{\\\"number_of_shards\\\":1,\\\"number_of_replicas\\\":1}\")
```
Query received on spark *after* this PR fix (query unescaped):
```
CREATE INDEX cv ON mys3.default.http_logs (`status`, `day`, `size`, `request`) WITH (auto_refresh = true, refresh_interval="5 Minutes", checkpoint_location = 's3://test',index_settings = "{\"number_of_shards\":1,\"number_of_replicas\":1}")
```

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by noCharger** [comment]
let's include this in 0.3 after testing

**Comment by dai-chen** [comment]
> let's include this in 0.3 after testing

Because we've started working on 0.4 task, is this the last PR for 0.3? If not, I think we'd better cut 0.3 branch now?

**Review by dai-chen — APPROVED** [review]

**Comment by noCharger** [comment]
> > let's include this in 0.3 after testing
> 
> Because we've started working on 0.4 task, is this the last PR for 0.3? If not, I think we'd better cut 0.3 branch now?

Can we include this https://github.com/opensearch-project/opensearch-spark/pull/303? I will add one more commit later today.

**Comment by dai-chen** [comment]
> > > let's include this in 0.3 after testing
> > 
> > 
> > Because we've started working on 0.4 task, is this the last PR for 0.3? If not, I think we'd better cut 0.3 branch now?
> 
> Can we include this #303? I will add one more commit later today.

Sure. Will block other 0.4 PRs for now.

**Comment by noCharger** [comment]
> > > > let's include this in 0.3 after testing
> > > 
> > > 
> > > Because we've started working on 0.4 task, is this the last PR for 0.3? If not, I think we'd better cut 0.3 branch now?
> > 
> > 
> > Can we include this #303? I will add one more commit later today.
> 
> Sure. Will block other 0.4 PRs for now.

We can cut the branch and backport too.

**Review by noCharger — APPROVED** [review]

File: spark-sql-application/src/main/scala/org/apache/spark/sql/FlintJobExecutor.scala [hunk 1]
```diff
@@ -371,6 +380,7 @@ trait FlintJobExecutor {
     val startTime = System.currentTimeMillis()
     // we have to set job group in the same thread that started the query according to spark doc
     spark.sparkContext.setJobGroup(queryId, "Job group for " + queryId, interruptOnCancel = true)
+    logInfo(s"Executing query: $query")
```
[the remark this answers is not in the document]
Comment (dai-chen) [reply spark-sql-application/src/main/scala/org/apache/spark/sql/FlintJobExecutor.scala:383]: @seankao-az @noCharger could you address this in other 0.3 PR if any?

**Pull request merged**